### ISSUE TITLE
Complete bookdown Documentation

### ISSUE THREAD

**Issue by imadmali** [stan-dev/bayesplot#89]
@jgabry if it's okay with you I'm starting an issue to track the conversion of the bayesplot vignettes to the more user friendly bookdown-style documentation.
- [x] Convert vignettes to bookdown
- [x] Use mc-stan.org CSS
- [x] Fix font options (CSS)
- [x] Fix figures
- [x] Fix broken links
- [x] Remove social media links
- [x] Include github link and links to other R interfaces
- [x] Include favicon
- [x] Custom blocks (warnings, notes, tips, etc.)

Notes:

* View current documentation at http://mc-stan.org/bayesplot/index.html
* Edit documentation in`book` directory in `gh-pages` branch
* `online-docs` branch is deprecated

**Comment by jgabry** [comment]
Yup, great!

**Comment by bob-carpenter** [comment]
Yes, that's great---bookdown renders so much more cleanly than knitr!  Any hints on knitr styles or alternatively how to strip down bookdown for smaller docs?

**Comment by imadmali** [comment]
@bob-carpenter I'm fairly new to it, but last week @jgabry and I managed to find a way to pull .R, .Rmd, etc. "chunks" into a bookdown doc and tinker with the CSS. The idea for this issue is to get a good template together that works for bayesplot and then implement that across the other R packages.

**Comment by imadmali** [comment]
If anyone wants to preview the current state of the documentation then go to https://htmlpreview.github.io/?https://raw.githubusercontent.com/stan-dev/bayesplot/online-docs/book/_book/index.html
I tried to match the mc-stan.org CSS as closely as possible.

**Comment by bob-carpenter** [comment]
That looks fantastic.  Thanks for rendering and putting the html up.   Three comments.

1. The nav's broken.  You can only click a subsection if you're in the section itself.

2.  Will there be a way to generate a pdf version of the doc?  Personally, I like to download pdf docs to my iPad and read there when I want to actually read docs (I'm a freak, I know), but then also have them online when I want to search.  

2.  As much as sans serif is a disaster for blocks of text (it's much harder to read than serifed and it looks so old fashioned), I think we should probably just follow what Hadley did for [_Advanced R_](http://adv-r.had.co.nz/Vocabulary.html), presumably to defend against users with low-res monitors where serif fonts don't render well.  On the other hand, the (_NY Times_)[http://www.nytimes.com] uses serif fonts.

**Comment by bob-carpenter** [comment]
Another question about the interface itself---is there going to be doc on how to use this from Stan or is this intended to be RStanArm only.  I looked at the first section on posterior predictive checks, which I'd like to use in my latest case study if I can do it directly in Stan.

**Comment by bob-carpenter** [comment]
I added a bunch of points to the comment a few issues back if you're only looking at email.

**Comment by imadmali** [comment]
> 1. The nav's broken. You can only click a subsection if you're in the section itself.

It works for me. Rendering stuff via github can be kind of weird. The best way to check if everything works is to serve the doc locally:

1. Pull the branch and navigate to `book/`
2. Open `book.Rproj` in RStudio.
3. Serve locally with `bookdown::serve_book()`

Then all the features should work (like search, etc.).

> 2. Will there be a way to generate a pdf version of the doc? Personally, I like to download pdf docs to my iPad and read there when I want to actually read docs (I'm a freak, I know), but then also have them online when I want to search.

There should be a PDF button in the header that, once clicked, downloads a pdf version of the doc.

> 3. As much as sans serif is a disaster for blocks of text (it's much harder to read than serifed and it looks so old fashioned), I think we should probably just follow what Hadley did for Advanced R, presumably to defend against users with low-res monitors where serif fonts don't render well. On the other hand, the (NY Times)[http://www.nytimes.com] uses serif fonts.

I thought serif is the old fashioned typeface and sans-serif is the cleaner looking one. Are you suggesting that we don't let users switch between the two?

> 4. Lots and lots of warnings.

Are you referring to these types of warnings:
```
## Warning in grid.Call.graphics(L_text, as.graphicsAnnot(x$label), x$x, x
## $y, : font "Symbol" could not be found for family "serif"
```
If so then this is something weird that happens on my machine when I use some of the bayesplot functions. I'm not sure how to fix that. Regardless, when @jgabry renders it there are no warnings. 

> 5. You want to make the top level title larger and/or bolder---the title needs to be easily perceived as the heaviest thing on the page.

Are you referring to the "bayesplot" title on the landing page or the `h1` section titles?

For your second comment, I'm not sure what you mean by "is there going to be doc on how to use this from Stan or is this intended to be RStanArm only."

**Comment by bob-carpenter** [comment]
Does the broken nav without their server mean that we won't be able to have a static HTML version of these pages---it'll need an Rstudio server?

The reason you get a lot of sans serif on computers is that until high-res displays, serif couldn't render finely enough.  When rendered properly, serifs make reading easier.  You'll note that nobody (other than Leimkuhler and Reich) sets an entire print book in sans serif.  That's because it's hard to read.  Other things that enhance readability:  reasonably short lines (60 to 80 characters max), not running text to margins, and generous, but not too generous inter-line spacing.

For titles, I meant the overall title on the landing page.  

The doc explains how to plot given a fit object from RStanArm.  So I'm asking if there'll be a way to run it from RStan with fit objects produced by `stan(...)`.  And if so, what needs to go in them.

**Comment by jgabry** [comment]
On Mon, May 22, 2017 at 1:21 PM, Bob Carpenter <notifications@github.com>
wrote:

> The doc explains how to plot given a fit object from RStanArm. So I'm
> asking if there'll be a way to run it from RStan with fit objects produced
> by stan(...). And if so, what needs to go in them.
>

Yeah there will be many examples of how to use bayesplot in conjunction
with RStan, not only RStanArm. It doesn't require you to use any particular
interface to Stan.

**Comment by imadmali** [comment]
> Does the broken nav without their server mean that we won't be able to have a static HTML version of these pages---it'll need an Rstudio server?

It's possible to have a static HTML version of these pages on github (see [here](https://bookdown.org/yihui/bookdown/github.html)) but I won't know whether the nav/search/etc. will work until I host it somewhere and test it out. @jgabry is is okay if I start a `gh-pages` branch to host the docs at http://stan-dev.github.io/bayesplot?

**Comment by jgabry** [comment]
On Mon, May 22, 2017 at 6:26 PM, Imad Ali <notifications@github.com> wrote:

> Does the broken nav without their server mean that we won't be able to
> have a static HTML version of these pages---it'll need an Rstudio server?
>
> It's possible to have a static HTML version of these pages on github (see
> here <https://bookdown.org/yihui/bookdown/github.html>) but I won't know
> whether the nav/search/etc. will work until I host it somewhere and test it
> out. @jgabry <https://github.com/jgabry> is is okay if I start a gh-pages
> branch to host the docs at http://stan-dev.github.io/bayesplot?
>
>
@imadmali Yeah go for it. Can't hurt to try it.

**Comment by imadmali** [comment]
Ok cool. Taking a quick glance at http://stan-dev.github.io/bayesplot, I think we're good on the navbar issue that @bob-carpenter flagged and the search issue.

**Comment by jgabry** [comment]
I'll be able to take a more thorough look tomorrow, but seems ok. Thanks
for working on this!

**Comment by imadmali** [comment]
@bob-carpenter going back to serif vs sans-serif... [the user can now choose](http://mc-stan.org/bayesplot/) between having a serif font or a sans-serif font. The sans-serif font is whatever I pulled from mc-stan.org. Do you have a recommendation of a good serif font (preferably something at https://fonts.google.com/)?

**Comment by bob-carpenter** [comment]
Don't you need to choose something that will render on client computers?

Whatever the NYTimes uses is good.  I'd just try Times Roman and
change it later if you need to.

- Bob

**Comment by jgabry** [comment]
@imadmali So is the gh-pages branch the right place to edit stuff now?

**Comment by jgabry** [comment]
If so, I can regenerate the content from my machine to avoid those font warnings you're getting

**Comment by jgabry** [comment]
@imadmali Also, have you been using `bookdown::html_book()` or `bookdown::html_document2` or some other function to regenerate all the files?

**Comment by imadmali** [comment]
Yep, gh-pages is the right branch to work out of.

Here's the steps I use to generate the documentation:
1. From `book.Rproj` run `render_book('index.Rmd', 'all')`.
2. Once all the rendering is done then run `sh cleanup.sh` from the terminal to move the contents of `book/bayesplot-output` into the root directory of the bayesplot repo (or do this manually).

If you're tinkering with rmd files and want a live preview of your changes as you make them then serve the book with `serve_book()` instead.

Also, we still need to fix figures since they're kinda big, and check for any broken links.

**Comment by imadmali** [comment]
@jgabry if you've done what I mentioned then you should probably push your updates (even if it's not perfect). The documentation is live at http://mc-stan.org/bayesplot and it's probably best not to have documentation out in the open with tons of warning messages.

**Comment by jgabry** [comment]
Ok I'll do it tonight.

**Comment by jgabry** [comment]
@bob-carpenter @imadmali Regarding the serif vs sans-serif issue, with **bayesplot** you can also toggle this using R's `options()`:

```r
options(bayesplot.base_family = "sans")
options(bayesplot.base_family = "serif")
```

**Comment by imadmali** [comment]
Awesome. I fixed a couple of things (title font boldness in main theme, subsection headings, giving installation it's own section). The pdf download icon somehow dropped, and I think it must be a weird yaml front matter arrangement thing in `index.Rmd`. I rearranged some of the options and then the icon came back. (@jgabry these changes won't be live until you render/move the site files.)

A few points (these aren't a big deal, but I think it'd be nice to have consistency across the R interfaces):

* We should specify an author (I put "Stan Development Team").
* If a package name is in the heading then it should either be title case *or* their "CRAN name". (I think title case looks better - especially when you download the pdf version of the doc.) I noticed that with [loo](http://mc-stan.org/interfaces/bayesplot.html) and [rstanarm](http://mc-stan.org/interfaces/rstanarm.html) we're using title case and with [bayesplot](http://mc-stan.org/interfaces/bayesplot.html) we use the package name.
* Right now the path to the page is http://mc-stan.org/bayesplot.html but the interface page is located at http://mc-stan.org/interfaces/bayesplot.html. I don't know if we want the documentation to be available at http://mc-stan.org/interfaces/bayesplot/documentation.html (broken link) or something like that.

**Comment by imadmali** [comment]
I think I've implemented all the important stuff, including some custom blocks that I think might be useful (see images below). Unless anyone has further suggestions (@betanalpha, @syclik you guys were most vocal in the meeting), I think this issue can be closed and we can roll this style of documentation out for the other R interfaces. 

(For people just joining the thread) To look at the latest version of the documentation:
1. Pull the `gh-pages` branch.
2. Open `book/book.Rproj` (via RStudio).
3. Run `bookdown::serve_book(getwd())`.
4. Tinker with Rmd files in `book/` for live updates (If you change the `book/style.css` go back to step 3).

![image](https://cloud.githubusercontent.com/assets/16767381/26470250/fd8b8d40-416a-11e7-97a2-95d8e43651b8.png)

![image](https://cloud.githubusercontent.com/assets/16767381/26470305/342a2b18-416b-11e7-9ff9-c13f85095ff2.png)

**Comment by bob-carpenter** [comment]
The problem with the switching to and from serifed fonts is that the
whole style is set up for that sans serif font.  The Times or whatever
is there is unreadable and the lines are way too wide and the interline
spacing way too large.  I wouldn't worry about it---looks like there
won't be much running text anyway and the sans serif looks OK.

For some reason, sections 1.0.1, 1.0.2, and 1.1 are all on the same
level in the left nav---I don't know why there's not a 1.0, but
it confused me looking at it.

I realized I was just confused by the titling and numbering and you
do go on and explain how to use posterior predictive checks later,
it's just that you start with an rstanarm example that originally led
me to believe I needed to do that.  

- Bob

**Comment by imadmali** [comment]
I fixed the section numbering if you render locally (along with some other things). I don't want my rendered version of the doc to be online since I get all those annoying font warnings. When @jgabry  renders/pushes my updates then everything should be fixed online.

**Comment by tjmahr** [comment]
Can we rename this thread to say _pkgdown_ instead of _bookdown_?

**Comment by jgabry** [comment]
We were originally playing around with bookdown so this whole thread might be obsolete at this point now that @imadmali switched us to pkgdown. @imadmali Can this be closed? If there are pkgdown-related issues we could start a new thread.

**Comment by jgabry** [comment]
I'm going to close this issue as the sites are now up for all of our R packages. We can open new issues for particular things we want to change.
